variable.c

```diff
@@ -134,6 +134,71 @@ rb_mod_name(VALUE mod)
     return classname(mod, &permanent);
 }
 
+/*
+ *  call-seq:
+ *     mod.set_temporary_name(string) -> self
+ *     mod.set_temporary_name(nil) -> self
+ *
+ *  Sets the temporary name of the module +mod+. This name is used as a prefix
+ *  for the names of constants declared in +mod+. If the module is assigned a
+ *  permanent name, the temporary name is discarded.
+ *
+ *  After a permanent name is assigned, a temporary name can no longer be set,
+ *  and this method raises a RuntimeError.
+ *
+ *  If the name given is not a string or is a zero length string, this method
+ *  raises an ArgumentError.
+ *
+ *  If the given name is +nil+, the module becomes anonymous.
+ *
+ *  Example:
+ *
+ *    m = Module.new # => #<Module:0x0000000102c68f38>
+ *    m.name #=> nil
+ *
+ *    m.set_temporary_name("M") # => M
+ *    m.name #=> "M"
+ *
+ *    m.set_temporary_name(nil) # => #<Module:0x0000000102c68f38>
+ *    m.name #=> nil
+ *
+ *    n = Module.new
+ *    n.set_temporary_name("Foo")
+ *
+ *    n::M = m
+ *    n::M.name #=> "Foo::M"
```

Comment:
Would it be possible to adjust the examples/documentation so that it doesn't use temporary names that look like constants? So maybe "(m)" instead of "M", and "foo" instead of "Foo", or something like that.

The normal usage of this feature is not to give constant-like names; for that you should just assign to a constant. It would be good if the first example that people see of the feature is not one that "lies" about the class name the way @eregon is worried about.

Reply:
I renamed all of them to `fake_name` which I think should be sufficient to address this concern.